### ISSUE TITLE
GM_xmlhttpRequest(...) usin' bad parameter for response parser call

### ISSUE THREAD

**Issue by trumpeta** [Tampermonkey/tampermonkey#1389]
(Please fill out the issue template with your details)

## Expected Behavior
Parses the requested responseType content

## Actual Behavior
tries to parse but uses invalid parameter for parser

The error is   `GET: TypeError: DOMParser.parseFromString: 'text/html; charset=UTF-8' (value of argument 2) is not a valid value for enumeration SupportedType`

Apparently TM is picking the second parameter for DOMParser.parseFromString from response header "Content-Type" which varies site by site, if returns "text/html", the parsing goes good, but if returns "text/html; charset=UTF-8", the response getter function reports an error and returns nothing. This should be fixed and used fixed argument "text/html" for the built in parser, or strip all parameters from received header string.

## Specifications

- FF 95.0.1
- TM: 4.13.4136
- OS: Win 8.1/x64

## Script

(Please give an example of the script if applicable.)

```js
// ==UserScript==
// @name BOO
// @namespace http://mywebsite.com
// @Version 1.0
// @description try to take over the world!
// @author Me
// @match *
// @grant GM_xmlhttpRequest(localParams
// @connect www.google.com
// ==/UserScript==

GM_xmlhttpRequest({
  method: 'GET',
  url: 'https://www.google.com/',
  responseType: 'document',
  onload: response => { console.debug(response.response) },
 });
```

**Comment by derjanb** [comment]
Fixed at the BETA version 4.13.4137 and above. 😅

**Comment by derjanb** [comment]
Duplicate of #1000
